Desktop_Interface/functiongencontrol.h

```diff
@@ -17,37 +17,42 @@ class functionGenControl : public QLabel
 {
     Q_OBJECT
 public:
-
 	struct ChannelData
 	{
 		std::vector<uint8_t> samples;
-		// TODO: get rid of length member. samples:std::vector already has this information
-		int length;
 		int divisibility;
 		double freq = 1000.0;
 		double amplitude = 0.0;
 		double offset = 0.0;
 	};
 
+	enum class ChannelID
```

Comment:
This is a really good thing that I think makes sense to use in many, many more places in the code.  For now, leave it here, but I think in future putting it in one of the global includes would be a good idea.